model/DataObject.php

```diff
@@ -3403,6 +3404,29 @@ public function databaseIndexes() {
 			}
 		}
 
+		if ($sort && is_string($sort)) {
+		    $sort = preg_split('/,(?![^()]*+\\))/', $sort);
```

Comment:
What's the significance of `*+` here?

Reply:
It's copied from `SQLselect::getsortorder` so I don't know :)

Reply:
Oh! I see... yeah that code itself isn't so easy to understand either. Apologies for implying it was your issue. :)

That code exists as-is in 4.0 too.

I think the SQLSelect::getOrderBy() only needs to return the SQL fragments that'll be sent to the final query; What we want is more heavily parsed, so I guess the ArrayList code is probably closer to our needs in this case?.

Oh well, once you step in a poop you're walking around in it all day aren't you. :)